setup.py

```diff
@@ -318,555 +340,566 @@ def find_directories(self, location, use_pkgconfig=False):
+            # reasons
+            if package.tag not in ["LZO", "LZO2"]:
+                dll_file = _platdep[package.tag][1] + ".dll"
                 # If DLL is not in rundir, do nothing.  This can be useful
                 # for BZIP2, that can be linked either statically (.LIB)
-                # or dinamically (.DLL)
+                # or dynamically (.DLL)
                 if rundir is not None:
-                    dll_files.append(os.path.join(rundir, dll_file))
+                    dll_files.append(rundir / dll_file)
 
-        if os.name == 'nt' and package.tag in ['HDF5']:
+        if os.name == "nt" and package.tag in ["HDF5"]:
             # hdf5.dll usually depends on zlib.dll
-            z_lib_path = ctypes.util.find_library('zlib.dll')
+            z_lib_path = ctypes.util.find_library("zlib.dll")
             if z_lib_path:
-                print('* Adding zlib.dll (hdf5 dependency): ``%s``' % z_lib_path)
-                dll_files.append(z_lib_path)
+                print(f"* Adding zlib.dll (hdf5 dependency): ``{z_lib_path}``")
+                dll_files.append(Path(z_lib_path))
 
-        if package.tag == 'LZO2':
+        if package.tag == "LZO2":
             lzo2_enabled = True
 
-    if lzo2_enabled:
-        lzo_package = lzo2_package
-    else:
-        lzo_package = lzo1_package
+    lzo_package = lzo2_package if lzo2_enabled else lzo1_package
 
     # ------------------------------------------------------------------------------
 
     cython_extnames = [
-        'utilsextension',
-        'hdf5extension',
-        'tableextension',
-        'linkextension',
-        '_comp_lzo',
-        '_comp_bzip2',
-        'lrucacheextension',
-        'indexesextension',
+        "utilsextension",
+        "hdf5extension",
+        "tableextension",
+        "linkextension",
+        "_comp_lzo",
+        "_comp_bzip2",
+        "lrucacheextension",
+        "indexesextension",
     ]
 
-
     def get_cython_extfiles(extnames):
-        extdir = 'tables'
+        extdir = ROOT / "tables"
         extfiles = {}
 
         for extname in extnames:
-            extfile = os.path.join(extdir, extname)
-            extpfile = '%s.pyx' % extfile
-            extcfile = '%s.c' % extfile
+            extfile = extdir / extname
+            extpfile = extfile.with_suffix(".pyx")
+            extcfile = extfile.with_suffix(".c")
 
-            if not exists(extcfile) or newer(extpfile, extcfile):
+            if not extcfile.exists() or newer(extpfile, extcfile):
                 # This is the only place where Cython is needed, but every
                 # developer should have it installed, so it should not be
                 # a hard requisite
                 from Cython.Build import cythonize
-                cythonize(extpfile, language_level = "2")
+
+                # TODO: language_level="3" ?
+                cythonize(str(extpfile), language_level="2")
```

Comment:
Recently someone told me that he has been able to successfully build PyTables with language_level="3".
Anyway I think that changing"language_level" should be done in a separate PR and carefully tested.
Also we probably need to bump the minimum Cython version required.

Reply:
I agree with separate PR and tests.